### PR TITLE
Fix queue constructor

### LINKED ISSUE #42: Constructor vs Additions
Just spotted in the docs you send a PR for. In the queue example we have this:

```php
fruits = new Queue(['apple', 'pear', 'peach', 'pine']);
$fruits->peek(); // return 'apple'
```

We also have this:

```php
$fruits = new Queue();
$fruits->add('apple', 'pear', 'peach', 'pine');
$fruits->peek(); // ???
```

What I'm saying, that constructor and addition methods gone out of sync in terms of mechanics:

```php
fruits = new Queue(['apple', 'pear', 'peach', 'pine']);
$fruits->add('apple', 'pear', 'peach', 'pine');
```

first time an array, second time a "spread array". It is a php5 articfact carried over.

What about something like this constructor:

```php
class Queue {
	constructor(...$elements) {
		$this->enqueue(...$elements);
	}
}
```

For `Map` it must stay as is.

It's tricky, WDYT?

### PR DESCRIPTION
**Pull request by cristianoc72**
I'm awfully sorry to come back on issue #38 but this commit is the only way I can find to make Queue work: enqueue at the end and poll from the top.

In this way, all tests are ok and the manipulations on the array, returned from `toArray` function, are the same as using the collection (see QueueTest::testMap).

@gossi please, test the code and share those examples that don't work as expected

This PR fixes #42

### CODE REVIEW

**Comment by gossi** [comment]
I must test this code on my machine, will earliest get to it on the weekend. The rest of the code should stay intact. If customizations need to be made, then in the constructor. An idea might be:

```php
class Queue {
	constructor(...$elements) {
		$this->enqueue(...array_reverse($elements));
	}
}
```

So, reversing the array at first, because the elements are unshifted one-by-one.

-- OR --

which (after re-reading php docs), is actually even better (and should be more performant):

```php
class Queue {
	enqueue(...$elements) {
		array_unshift($this->array, ...$elements);
		return $this;
	}
}
```

Dunno if this will change the current behavior but I'd rather would go for this approach, actually.

WDYT?

File: src/collection/Queue.php [hunk 1]
```diff
@@ -40,7 +38,7 @@ public function __construct($collection = []) {
 	 */
 	public function enqueue(...$elements): self {
 		foreach ($elements as $element) {
-			array_unshift($this->array, $element);
+			array_push($this->array, $element);
```
Comment (gossi) [inline src/collection/Queue.php]: I think this methods body will just be `array_unshift($this->array, ...$elements)` (without the loop)

is it?

Comment (cristianoc72) [reply src/collection/Queue.php]: Fixed. Now tests are ok.

**Comment by cristianoc72** [comment]
Now there is no "strange" behavior and Queue correctly works with `array_shift` and `array_unshift`.

**Review by gossi — APPROVED** [review]

**Comment by gossi** [comment]
that's great. Thank you so much. Merge for honor ;)

**Comment by cristianoc72** [comment]
> Merge for honor ;)

:rofl:

**Pull request merged**